specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+
+TODO: why the string value, and not an integer?   
+TODO: no double underscore guard for attributes likely makes sense
+
+### Function Body
+
+The function body consists of a single block named `entry`. This implies in particular that no branching is permitted inside a function's body.
+In the `entry` block, any number of calls to QIS functions may be performed.
+To be compatible with the Base Profile these functions must return void. 
+Any arguments to invoke them must be inlined into the call itself; 
+they hence must be constants or a pointer to a [qubit or result]()
+The result of the program execution should be logged using output recording functions.
+
+The following instructions are the *only* LLVM instructions that are permitted within a Base Profile compliant program:
+
+| LLVM Instruction                  | Context and Purpose         | Rules for Usage |
+|---------------------------|-------------------|-------------|
+| `call`    | Used within a function block to invoke any one of the declared QIS functions and the output recording functions. | May optionally be preceded by a [`tail` marker](https://llvm.org/docs/LangRef.html#call-instruction). |
+| `ret` | Used to return the exit code of the program. | Must occur (only) as the final instruction of the `entry` block. |
+| `inttoptr` | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`. | May be used as part of a function call only. |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value. | May be used as part of call to an output recording function only. |
+
+## Data Types and Values
+
+Within the base profile, defining local variables is not supported. Arguments to calls correspondingly are expected to be constant values. 
+
+Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*` respectively, where the pointer itself rather than the memory location identifies the qubit or result: For the purpose of passing them as arguments in function calls, a 64-bit signed integer value is cast to the appropriate pointer type. The signed integer that is case must be in the interval `[0, nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and `nrResults` are the required number of qubits and results defined by entry point attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all backends. 
+
+Integers and double-precision floating point numbers are available as in full QIR;
+however, computations using these numeric types are not available.
```

Comment:
At least from the perspective of how things currently work: Yes, we have support for literals of any type (at the moment) such that you could have `qis` functions that would take an angle, tolerance etc. in addition to quantum registers. It is the expectation that lowering of the IR to the backends machine code will take care of interpreting the literal and translate it correctly. 

One thing to note though: If the base profile is to represent the common subset of capabilities, then literals are effectively not supported since the [candidate functions](https://github.com/qir-alliance/qat/blob/main/qir/qat/ValidationPass/ValidationPassConfiguration.cpp) of `provider_b340` does not take any argument that is a primitive type. Therefore, while you are allowed to write a literal, you have no place to put.

Reply:
Let's carefully distinguish profile from instruction set. The document is now fleshed out a lot further, and I hope I made it clear that basically you can have constants of any type but only as arguments to calls or as the final integer to return. Hence, what constant values can possibly occur is effectively determined by the chosen QIS.